overlord/state/state.go

```diff
@@ -241,6 +241,26 @@ func (s *State) EnsureBefore(d time.Duration) {
 // ErrNoState represents the case of no state entry for a given key.
 var ErrNoState = errors.New("no state entry for key")
 
+// NoStateError represents the case where no state could be found for a given key.
+type NoStateError struct {
+	// Key is the key for which no state could be found.
+	Key string
+}
+
+func (e *NoStateError) Error() string {
+	var keyMsg string
+	if e.Key != "" {
+		keyMsg = fmt.Sprintf(" %q", e.Key)
+	}
+
+	return fmt.Sprintf("no state entry for key%s", keyMsg)
+}
+
+func (e *NoStateError) Is(err error) bool {
+	_, ok := err.(*NoStateError)
+	return ok || err == ErrNoState // nocheck=
```

Comment:
(niptick) When I read this for the first time I had the feeling that something is missing here, i.e. that there is something after the "=" that should be added. After reading it again I guess it means `nocheck-for-equals`. I was playing around a bit but couldn't come up with something obviously better, I tried: `no-check=equal`, `nocheck` `disable-check` but it is super nitpicky so feel free to just ignore.